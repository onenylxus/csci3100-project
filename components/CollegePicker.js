// Import
import React from 'react';
import { Text, View, Dimensions } from 'react-native';
import { Picker } from '@react-native-community/picker';
import CollegeList from '../assets/json/collegeList.json';
import Style from '../assets/style';

// Export college picker
export default function CollegePicker({ callback, value }) {
  const windowWidth = Dimensions.get('window').width;

  function styleByDevice(widthOfDevice, component) {
    if (widthOfDevice < 1100) {
      // Style of Small Screen
      switch (component) {
        case 'width':
          return '100%';

        case 'pickerWidth':
          return '100%';

        default:
          break;
      }
    }

    // Style of Large Screen
    switch (component) {
      case 'width':
        return 550;

      case 'pickerWidth':
        return '75%';

      default:
        break;
    }
  }

  return (
    <View>
      {/* College */}
      <View
        style={{
          marginVertical: 40,
          flexDirection: 'column',
          width: styleByDevice(windowWidth, 'width'),
        }}
      >
        <Text style={Style.pickerTitle}>College: </Text>
        <View>
          <Picker
            style={{
              width: styleByDevice(windowWidth, 'pickerWidth'),
              alignSelf: 'center',
            }}
            selectedValue={value}
            onValueChange={(itemValue) => callback(itemValue)}
          >
            {[
              <Picker.Item label="Choose a college" value="" key="" />,
              ...Object.keys(CollegeList).map((item) => (
                <Picker.Item
                  label={CollegeList[item]}
                  value={item}
                  key={item}
                />
              )),
            ]}
          </Picker>
        </View>
      </View>
    </View>
  );
}
