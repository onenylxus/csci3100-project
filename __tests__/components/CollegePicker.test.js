// Import
import React from 'react';
import { render } from '@testing-library/react-native';

// Import target component
import CollegePicker from '../../components/CollegePicker';

// Run jest tests
describe('CollegePicker', () => {
  let element;

  beforeEach(() => {
    // Render
    element = render(<CollegePicker />);
  });

  it('matches snapshot', () => {
    expect(element.toJSON()).toMatchSnapshot();
  });
});
