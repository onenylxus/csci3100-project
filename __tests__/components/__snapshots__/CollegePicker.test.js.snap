// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollegePicker matches snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "marginVertical": 40,
        "width": "100%",
      }
    }
  >
    <Text>
      College: 
    </Text>
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "width": "100%",
          }
        }
      >
        <RNCPicker
          items={
            Array [
              Object {
                "label": "Choose a college",
                "textColor": undefined,
                "value": "",
              },
              Object {
                "label": "Chung Chi College",
                "textColor": undefined,
                "value": "CC",
              },
              Object {
                "label": "New Asia College",
                "textColor": undefined,
                "value": "NA",
              },
              Object {
                "label": "United College",
                "textColor": undefined,
                "value": "UC",
              },
              Object {
                "label": "Shaw College",
                "textColor": undefined,
                "value": "SC",
              },
              Object {
                "label": "Morningside College",
                "textColor": undefined,
                "value": "MC",
              },
              Object {
                "label": "S.H. Ho College",
                "textColor": undefined,
                "value": "SH",
              },
              Object {
                "label": "CW Chu College",
                "textColor": undefined,
                "value": "WC",
              },
              Object {
                "label": "Wu Yee Sun College",
                "textColor": undefined,
                "value": "YS",
              },
              Object {
                "label": "Lee Wo Sing College",
                "textColor": undefined,
                "value": "WS",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
